### PR TITLE
Improvement: Correct type annotation for azure_ad_token_provider Closed: #14402

### LINKED ISSUE #14402: AzureOpenAIEmbeddings cannot authenticate with azure_ad_token_provider
### System Info

Langchain: 0.0.346
OpenAI:  1.3.7

### Who can help?

_No response_

### Information

- [ ] The official example notebooks/scripts
- [X] My own modified scripts

### Related Components

- [ ] LLMs/Chat Models
- [X] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [ ] Document Loaders
- [ ] Vector Stores / Retrievers
- [ ] Memory
- [ ] Agents / Agent Executors
- [ ] Tools / Toolkits
- [ ] Chains
- [ ] Callbacks/Tracing
- [ ] Async

### Reproduction

Simple script to authenticate to Azure with RBAC

```
from langchain.embeddings import AzureOpenAIEmbeddings
from azure.identity import DefaultAzureCredential, get_bearer_token_provider

token_provider = get_bearer_token_provider(DefaultAzureCredential(), "https://cognitiveservices.azure.com/.default")
embeddings = AzureOpenAIEmbeddings(azure_endpoint='xxxxxxx', azure_ad_token_provider=token_provider)
```

### Expected behavior

Should authenticate, but is seems like the `azure_ad_token_provider` is not added to the values dict

langchain/embeddings/azure_openai.py line 80-86
```
values["azure_endpoint"] = values["azure_endpoint"] or os.getenv(
            "AZURE_OPENAI_ENDPOINT"
        )
values["azure_ad_token"] = values["azure_ad_token"] or os.getenv(
            "AZURE_OPENAI_AD_TOKEN"
        )
```
Other parameters are added to values, but not `azure_ad_token_provider`

### PR DESCRIPTION
**Pull request by 169**
Description
Fix https://github.com/langchain-ai/langchain/issues/14402, Similar changes: https://github.com/langchain-ai/langchain/pull/14166

Twitter handle
[lin_bob57617](https://twitter.com/lin_bob57617)

### CODE REVIEW

**Review by eyurtsev — APPROVED** [review]

**Comment by eyurtsev** [comment]
@baskaryan good to merge whenever you're ready to sequence it

File: libs/langchain/langchain/embeddings/azure_openai.py [hunk 1]
```diff
@@ -38,7 +38,7 @@ class AzureOpenAIEmbeddings(OpenAIEmbeddings):
         For more: 
         https://www.microsoft.com/en-us/security/business/identity-access/microsoft-entra-id.
     """  # noqa: E501
-    azure_ad_token_provider: Union[str, None] = None
+    azure_ad_token_provider: Union[Callable[[], str], None] = None
```
Comment (mspronesti) [inline libs/langchain/langchain/embeddings/azure_openai.py:41]: Is this only relevant for `AzureOpenAIEmbeddings`? Doesn't the same problem exist in `AzureOpenAI` completion and chat models?

Comment (169) [reply libs/langchain/langchain/embeddings/azure_openai.py:41]: Yes, it has been partially fixed before, ref: https://github.com/langchain-ai/langchain/pull/14166

**Comment by 169** [comment]
@baskaryan, @eyurtsev, @hwchase17 I have changed it to the code in langchain_community, PTAL

**Review by darioxz — APPROVED** [review]

**Review by efriis — APPROVED** [review]

**Pull request merged**